src/main/java/school/faang/bjs2_74534/Main.java

```diff
@@ -0,0 +1,36 @@
+package school.faang.bjs2_74534;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+@Slf4j
+public class Main {
+    private static final String TASK_RESULT = "{} has completed the quest and now has {} experience points.";
+
+    public static void main(String[] args) {
+        QuestSystem questSystem = new QuestSystem();
+
+        Player firstPlayer = new Player("Thrall", 10, 250);
+        Player secondPlayer = new Player("Sylvanas", 12, 450);
+
+        Quest firstQuest = new Quest("Defeat the Lich King", 10, 150);
+        Quest secondQuest = new Quest("Retrieve the Sword of Azeroth", 8, 100);
+
+        List<CompletableFuture<Player>> workingQuest = List.of(
+            questSystem.startQuest(firstPlayer, firstQuest),
+            questSystem.startQuest(firstPlayer, secondQuest),
+            questSystem.startQuest(secondPlayer, secondQuest)
+        );
+
+        workingQuest.forEach(playerQuest -> playerQuest.thenAccept(Main::acceptInfo));
```

Comment:
советую использовать другой подход:

```suggestion
       workingQuest.stream()
                .peek(future -> future.thenAccept(questSystem::onQuestCompletion))
                .map(CompletableFuture::join)
                .collect(Collectors.toSet());
```

Reply:
я рассматривал этот вариант, но мне не понравилось, что ожидая получения результата всех потоков  в метод QuestSystem.onQuestCompletion приходит объект Player с ИТОГОВЫМ результатом выполнения обоих потоков. Т.е в 21 строке я запускаю два процесса по одному игроку "firstPlayer = Thrall". После завершения процессов я ожидал получить сообщения вида (то, что сейчас закомментировано в строках 27 и 28):
```
Thrall has completed the quest and now has 350 experience points.
Thrall has completed the quest and now has 500 experience points.
```
, где первая строка результат выполнения первого процесса, а вторая строка результат выполнения второго.

а в твоём варианте мы имеем:
```
> Thrall has completed the quest and now has 500 experience points.
> Thrall has completed the quest and now has 500 experience points.
```

Reply:
Справедливо

Reply:
а так можно? )))